pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
I may not have clearly communicated what I meant.  I don't think this method should know anything about watch.EventTypes.  Instead, when you are printing out from watch, you have a top level printer EventPrinter that invokes the ResourcePrinter for each object that comes through the watch (the `event.Object`).  You invoke the ResourcePrinter with a object from the watch (say a Pod) and it should only print the columns a Pod has.  Then there has to be some way for the wrapping EventPrinter to append the event column.  One way is to, instead of using the generic ResourcePrinter (which deals with a writer), create a new `ColumnResourcePrinterFunc func(runtime.Object) ([][]string, error)`, and invoke that.  Another way is to pass a "PrependingWriter" that adds a new column to the front of any line.

I'd really like the printer functions to better represent what they do so they can be composed more easily.  The `io.Writer` as a target loses all the structure - so I tend to think the ColumnResourcePrinter is a better factoring for our code today (which can easily be adapted to a ResourcePrinter).